### PR TITLE
[`Frontend`]: Explore page

### PR DESCRIPTION
**Pull request by tokitouq**

### CODE REVIEW

File: backend/django_core/static_src/components/page/index.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    { tagname: `explore`, component: await import("./explore/Index.svelte") },
+]
+
+mapping.forEach((item) => {
+    svelteRetag({
+        component: item.component.default,
+        tagname: `coreproject-page-${item.tagname}`,
+        attributes: true,
+        shadow: false,
+        hydratable: true
```
Comment (baseplate-admin) [inline backend/django_core/static_src/components/page/index.ts:15]: This should be `false`

File: backend/django_core/templates/anime/explore/index.html [hunk 2]
```diff
@@ -1 +1 @@
-<h1>Explore</h1>
+<coreproject-page-explore></coreproject-page-explore>
```
Comment (baseplate-admin) [inline backend/django_core/templates/anime/explore/index.html:1]: Add `<meta/>` tags

File: backend/django_core/static_src/components/page/explore/Index.svelte [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+                                parent_class="md:max-h-[30vw] bg-neutral w-full"
+                            >
+                                {#each Object.entries(filter_items) as item}
+                                    {@const key = item[0]}
+                                    {@const value = item[1]}
+
+                                    {@const is_selected = selected_items.some(selected_item => selected_item[0] === key)}
+
+                                    <button
+                                        on:click={() => update_selected_items(option[0], item)}
```
Comment (baseplate-admin) [inline backend/django_core/static_src/components/page/explore/Index.svelte:198]: Add a `preventDefault` modifier

File: backend/django_core/static_src/components/page/explore/Index.svelte [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+                                        class="relative btn btn-neutral h-max min-h-max leading-none flex items-center justify-start md:rounded-[0.35vw] p-3 text-sm md:px-[1vw] md:py-[0.75vw] md:text-[0.9vw]">
+                                        {value}
+
+                                        {#if is_selected}
+                                            <div class="absolute right-[0.75vw] rounded-full bg-primary text-white p-1 md:p-[0.25vw]">
+                                                <Tick class="w-2 md:w-[0.75vw] text-white" />
+                                            </div>
+                                        {/if}
```
Comment (baseplate-admin) [inline backend/django_core/static_src/components/page/explore/Index.svelte:206]: maybe add animation

File: backend/django_core/static_src/components/page/explore/Index.svelte [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+                            >
+                                <coreproject-icon-cross class="md:w-[1vw]"></coreproject-icon-cross>
+                            </button>
+                        {:else}
+                            <button class="btn !bg-transparent border-none absolute right-0 mr-3 w-4 p-0 md:mr-[1vw] md:w-[1vw]">
+                                <coreproject-icon-chevron class="md:w-[1vw]"></coreproject-icon-chevron>
+                            </button>
+                        {/if}
+                    </div>
+                    <!-- svelte-ignore a11y-no-noninteractive-tabindex -->
```
Comment (baseplate-admin) [inline backend/django_core/static_src/components/page/explore/Index.svelte:181]: This should not be the case. Do you need to make it like this?

File: backend/django_core/static_src/components/page/explore/Index.svelte [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+                        {#if selected_items.length > 0}
+                            <button
+                                on:click={() => clear_selected_items(option[0])}
+                                class="btn !bg-transparent border-none absolute right-0 mr-3 w-4 p-0 md:mr-[1vw] md:w-[1vw]"
+                            >
+                                <coreproject-icon-cross class="md:w-[1vw]"></coreproject-icon-cross>
+                            </button>
+                        {:else}
+                            <button class="btn !bg-transparent border-none absolute right-0 mr-3 w-4 p-0 md:mr-[1vw] md:w-[1vw]">
+                                <coreproject-icon-chevron class="md:w-[1vw]"></coreproject-icon-chevron>
```
Comment (baseplate-admin) [inline backend/django_core/static_src/components/page/explore/Index.svelte:177]: Again you import direct `svelte` component inside `svelte` file

File: backend/django_core/static_src/components/page/explore/Index.svelte [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+                            type="text"
+                            tabindex="0" role="button"
+                            class="peer w-full rounded-lg border-none bg-neutral py-3 text-base leading-none placeholder focus:ring-0 md:w-[11vw] md:rounded-[0.5vw] md:bg-neutral text-neutral-content md:py-[0.8vw] md:pl-[1vw] md:text-[1vw] placeholder:font-medium font-semibold"
+                        />
+                        {#if selected_items.length > 0}
+                            <button
+                                on:click={() => clear_selected_items(option[0])}
+                                class="btn !bg-transparent border-none absolute right-0 mr-3 w-4 p-0 md:mr-[1vw] md:w-[1vw]"
+                            >
+                                <coreproject-icon-cross class="md:w-[1vw]"></coreproject-icon-cross>
```
Comment (baseplate-admin) [inline backend/django_core/static_src/components/page/explore/Index.svelte:173]: Again you import direct `svelte `component inside `svelte `file

File: backend/django_core/static_src/components/page/explore/Index.svelte [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+                        <input
+                            bind:value={option[1].value}
+                            on:blur={() => (option[1].value = "")}
+                            type="text"
+                            tabindex="0" role="button"
+                            class="peer w-full rounded-lg border-none bg-neutral py-3 text-base leading-none placeholder focus:ring-0 md:w-[11vw] md:rounded-[0.5vw] md:bg-neutral text-neutral-content md:py-[0.8vw] md:pl-[1vw] md:text-[1vw] placeholder:font-medium font-semibold"
+                        />
+                        {#if selected_items.length > 0}
+                            <button
+                                on:click={() => clear_selected_items(option[0])}
```
Comment (baseplate-admin) [inline backend/django_core/static_src/components/page/explore/Index.svelte:170]: Add a preventDefault modifier

File: backend/django_core/static_src/components/page/explore/Index.svelte [hunk 9]
```diff
@@ -0,0 +1,220 @@
...
+                                        +{selected_items.filter((item) => item !== selected_items[0]).length}
+                                    </span>
+                                {/if}
+                            {:else}
+                                <span class="ml-3 text-base md:ml-[1vw] md:text-[0.9vw]">Any</span>
+                            {/if}
+                        </span>
+                        <input
+                            bind:value={option[1].value}
+                            on:blur={() => (option[1].value = "")}
```
Comment (baseplate-admin) [inline backend/django_core/static_src/components/page/explore/Index.svelte:163]: This should be a `<svelte:window/>` modifier

File: backend/django_core/static_src/components/page/explore/Index.svelte [hunk 10]
```diff
@@ -0,0 +1,220 @@
...
+        // update filter_options_mapping
+        filter_options_mapping[key].selected_items = [];
+    };
+</script>
+
+<section class="mt-20 flex flex-col p-5 md:mt-0 md:gap-[1.5vw] md:pb-[2.5vw] md:pl-[1.5vw] md:pr-[3.75vw] md:pt-0">
+    <div class="flex flex-col gap-2 md:gap-[0.5vw]">
+        <span class="text-2xl font-bold leading-none md:text-[2vw]">
+            Anime <span class="text-warning">Explore</span>
+        </span>
```
Comment (baseplate-admin) [inline backend/django_core/static_src/components/page/explore/Index.svelte:107]: `<div>` then `<span/>` 

not

 `<span/>`  in  `<span/>`

File: backend/django_core/static_src/components/page/explore/Index.svelte [hunk 11]
```diff
@@ -0,0 +1,220 @@
...
+    import MoreBox from "$icons/MoreBox/Index.svelte";
+
+    // Mapping
+    let filter_options_mapping: {
+        [key: string]: {
+            title: string;
+            class: string;
+            value: string;
+            items?: Record<string, string> | undefined;
+            selected_items: Array<[string, string]>;
```
Comment (baseplate-admin) [inline backend/django_core/static_src/components/page/explore/Index.svelte:14]: `selected_items?: Array<[string, string]>;`

File: backend/django_core/static_src/components/page/explore/Index.svelte [hunk 12]
```diff
@@ -0,0 +1,220 @@
...
+    import Tick from "$icons/Tick/Index.svelte";
+    import MoreBox from "$icons/MoreBox/Index.svelte";
+
+    // Mapping
+    let filter_options_mapping: {
+        [key: string]: {
+            title: string;
+            class: string;
+            value: string;
+            items?: Record<string, string> | undefined;
```
Comment (baseplate-admin) [inline backend/django_core/static_src/components/page/explore/Index.svelte:13]: items are not `undefined`, it should be `null`

**Review by baseplate-admin — CHANGES_REQUESTED** [review]
change these please

File: backend/django_core/static_src/components/page/explore/Index.svelte (hunk 4 above)
Comment (tokitouq) [reply backend/django_core/static_src/components/page/explore/Index.svelte:211]: For what? this `.dropdown` has already a nice animation

File: backend/django_core/static_src/components/page/explore/Index.svelte (hunk 5 above)
Comment (tokitouq) [reply backend/django_core/static_src/components/page/explore/Index.svelte:186]: Yes? cause without `tabindex` this dropdown will close after clicking ( which we dont want like in v3 ).
OR we can use a clickable element like `button` that looks off :\

File: backend/django_core/static_src/components/page/explore/Index.svelte (hunk 9 above)
Comment (tokitouq) [reply backend/django_core/static_src/components/page/explore/Index.svelte:168]: No? this is supposed for clearing value when click outside that specific input.
Not on:blur window

File: backend/django_core/static_src/components/page/explore/Index.svelte (hunk 10 above)
Comment (tokitouq) [reply backend/django_core/static_src/components/page/explore/Index.svelte:110]: Okay, but its quite common to nest `span` tags.
Changed anyway :)

File: backend/django_core/static_src/components/page/explore/Index.svelte (hunk 11 above)
Comment (tokitouq) [reply backend/django_core/static_src/components/page/explore/Index.svelte:17]: We need to initialize an array anyway, ??
`filter_option.selected_items = [...filter_option.selected_items, selected_item];`

Oh you mean just this type? dont change mapping? Okay

File: backend/django_core/static_src/components/page/explore/Index.svelte (hunk 12 above)
Comment (tokitouq) [reply backend/django_core/static_src/components/page/explore/Index.svelte:16]: This `items` is already an optional type, (`undefined`), huh?

Comment (baseplate-admin) [reply backend/django_core/static_src/components/page/explore/Index.svelte]: It should still be null by default

File: backend/django_core/static_src/components/page/explore/Index.svelte (hunk 5 above)
Comment (baseplate-admin) [reply backend/django_core/static_src/components/page/explore/Index.svelte:186]: > we can use a clickable element like button that looks off :\

This is what i did

File: backend/django_core/static_src/components/page/explore/AnimeCard.svelte [hunk 13]
```diff
@@ -0,0 +1,124 @@
+<script lang="ts">
+    export let anime_name: string;
+    export let anime_image: string;
+    export let anime_total_episodes: number;
+    export let anime_synopsis: string;
+    export let anime_studio: string[];
+    export let anime_genres: string[];
+
+    // Pass styles
+    export let dropdown_class: string;
```
Comment (baseplate-admin) [inline backend/django_core/static_src/components/page/explore/AnimeCard.svelte:10]: This should not be used.

File: backend/django_core/static_src/components/page/explore/Index.svelte [hunk 14]
```diff
@@ -0,0 +1,373 @@
...
+            >
+                {#each trending_animes as anime}
+                    <AnimeCard
+                        anime_name={anime.name}
+                        anime_studio={anime.studios}
+                        anime_image={anime.cover}
+                        anime_genres={anime.genres}
+                        anime_synopsis={anime.synopsis}
+                        anime_total_episodes={anime.episodes_count}
+                        dropdown_class="md:pl-[1vw]"
```
Comment (baseplate-admin) [inline backend/django_core/static_src/components/page/explore/Index.svelte:367]: This too

**Review by baseplate-admin — CHANGES_REQUESTED** [review]
Remvoe these

File: backend/django_core/static_src/components/page/explore/Index.svelte [hunk 15]
```diff
@@ -22,7 +22,7 @@
     let result_animes_element: HTMLDivElement;
     let search_query = "";
 
-    const handle_input = async () => {
+    const handle_fetch = async () => {
```
Comment (baseplate-admin) [inline backend/django_core/static_src/components/page/explore/Index.svelte:25]: handle input better

**Review by baseplate-admin — CHANGES_REQUESTED** [review]

File: backend/django_core/templates/anime/explore/index.html [hunk 16]
```diff
@@ -1 +1,5 @@
-<h1>Explore</h1>
+{% block head %}
+    <title>Explore - AnimeCore</title>
```
Comment (baseplate-admin) [inline backend/django_core/templates/anime/explore/index.html:2]: Add opengraph tags

Comment (tokitouq) [reply backend/django_core/templates/anime/explore/index.html:2]: Tags? via svelte component?

Comment (baseplate-admin) [reply backend/django_core/templates/anime/explore/index.html:2]: NO SVELTE

**Pull request merged**